### PR TITLE
Multicoin support for ENS

### PR DESCRIPTION
**Pull request by Arachnid**
See discussion and EIP draft here: https://discuss.ens.domains/t/new-standard-proposal-ens-multicoin-support/1148/20

### CODE REVIEW

File: contracts/profiles/AddrResolver.sol [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 contract AddrResolver is ResolverBase {
     bytes4 constant private ADDR_INTERFACE_ID = 0x3b3b57de;
+    bytes4 constant private ADDRESS_INTERFACE_ID = 0xf1cb7e06;
+    uint constant private COIN_TYPE_ETH = 60;
 
     event AddrChanged(bytes32 indexed node, address a);
+    event AddressChanged(bytes32 indexed node, uint coinType, bytes newAddress);
 
-    mapping(bytes32=>address) addresses;
+    mapping(bytes32=>mapping(uint=>bytes)) _addresses;
```
Comment (decanus) [inline contracts/profiles/AddrResolver.sol]: Why is ETH 60?

File: contracts/profiles/AddrResolver.sol [hunk 2]
```diff
@@ -26,10 +28,26 @@ contract AddrResolver is ResolverBase {
...
     function addr(bytes32 node) public view returns (address) {
-        return addresses[node];
+        bytes memory a = addr(node, COIN_TYPE_ETH);
+        if(a.length == 0) {
+            return address(0);
+        }
+        return bytesToAddress(a);
+    }
+
+    function setAddr(bytes32 node, uint coinType, bytes memory a) public authorised(node) {
```
Comment (decanus) [inline contracts/profiles/AddrResolver.sol]: Would it make sense to have some form of Code Repository with codes to strings?

File: contracts/profiles/AddrResolver.sol (hunk 1 above)
Comment (Amxx) [reply contracts/profiles/AddrResolver.sol]: Addr id are to follow SLIP-0044 : https://github.com/satoshilabs/slips/blob/master/slip-0044.md

Comment (decanus) [reply contracts/profiles/AddrResolver.sol]: @amxx perfect thanks

File: contracts/profiles/AddrResolver.sol (hunk 2 above)
Comment (Arachnid) [reply contracts/profiles/AddrResolver.sol]: Definitely - but offchain.

**Pull request merged**